guides/v2.1/coding-standards/xml-coding-standard.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: default
+group: coding-standards
+subgroup: Coding standards
+title: Layout XML Coding Standards
+menu_title: Layout XML Coding Standards
+menu_order: 100
+contributor_name: SwiftOtter Studios
+contributor_link: https://swiftotter.com/
+version: 2.1
+github_link: coding-standards/xml-coding-standard.md
+---
+
+# Layout XML Coding Standard
+
+### Name attribute
+
+The `name` attribute should adhere to the guidelines listed below. The `name` attribute should:
+
+- Be unique to the project.
+- Use a namespace approach where each child block's name contains its parent's name as well.
+- Have segments separated by a `.` (period).
+- Be all lowercase.
+- Never use `_` (underscores).
+
+*Example:*
+
+```xml
+<block name="header">
+  <block name="header.right">
+    <block name="header.right.search">
+      <block name="header.right.search.autocomplete" />
+    </block>
+  </block>
+</block>
+```
+
+### As attribute
+
+The `as` attribute should follow the guidelines below. The value of `as`:
+
+- Only needs to be unique to the block that contains it.
```

Comment:
I have an internal goal of having this complete tonight (12-Jun), but it will be a couple more days. Thanks again for proposing.